src/history/html5.js

```diff
@@ -8,24 +8,37 @@ import { setupScroll, handleScroll } from '../util/scroll'
 import { pushState, replaceState, supportsPushState } from '../util/push-state'
 
 export class HTML5History extends History {
+  +initLocation: string
+
   constructor (router: Router, base: ?string) {
     super(router, base)
 
+    this.initLocation = getLocation(this.base)
+  }
+
+  setupListeners () {
```

Comment:
This is an important new behavior to call out. Previously HTML5History would set up listeners during its constructor, but in this PR it only sets up listeners once setupListeners is called.

My changes to index.js ensure that setupListeners() is called once an app is initialized. I am unsure if this class is ever used directly by anything else besides index.js, but if it is then there's a chance this is breaking behavior and we should consider a different implementation.